charts/openshift-metering/templates/openshift-reporting/report-queries/persistentvolumeclaim-usage.yaml

```diff
@@ -196,19 +196,32 @@ spec:
     unit: bytes
   inputs:
   - name: ReportingStart
+    type: time
   - name: ReportingEnd
+    type: time
   - name: PersistentvolumeclaimUsageWithPhaseRawDataSourceName
     type: ReportDataSource
     default: persistentvolumeclaim-usage-with-phase-raw
   query: |
+    WITH averaged_persistentvolumeclaim_usage_per_pvc AS (
+      SELECT
+        timestamp '{| default .Report.ReportingStart .Report.Inputs.ReportingStart| prestoTimestamp |}' AS period_start,
+        timestamp '{| default .Report.ReportingEnd .Report.Inputs.ReportingEnd | prestoTimestamp |}' AS period_end,
+        namespace,
+        persistentvolumeclaim,
+        avg(persistentvolumeclaim_usage_bytes) as avg_persistentvolumeclaim_usage_bytes
```

Comment:
Averaging out these usage values per PVC in a particular period is definitely a step in the right direction, especially given the reporting for this query is _way_ off, but I wonder if there's a better way to calculate usage over time using a more suitable aggregate function than `avg`.

That said, after testing this locally in various scenarios, this still holds up nicely and it's close to the actual PVC disk usage, but skews slightly behind in the case that usage is drastically increased in a short period of time ( e.g. large files are copied into the running container) but this may be the closest we can get without something more complex.

Reply:
I can't think of a more appropriate aggregate function than AVG. Yeah it will skew off for large data storage increases over short time, but the alternative is probably no longer something that can be readily accomplished with SQL and would instead head off in the direction of a programmatic analysis of the disk usage. I doubt this is what the cloud providers do for billing, instead likely choosing something very close to the SQL AVG usage to come up with a number to bill based on.